### DISCUSSION TITLE
Return other objects/arrays in getStaticPaths() function

### DISCUSSION THREAD

**Discussion by 0jg** [vercel/next.js#14150] [Help]
Hey,

I make an API call during the `getStaticPaths()` async to get a list of paths to return (here, journals.slug). During this API call, I would also like to pass other information such as {journals.name} to avoid repeated API calls.

Is this possible? `getStaticPaths()` only allows two values to be returned (else an error is thrown `(╯°□°)╯︵ [error]`).

Here's my code. I need a mechanism to also pass the `names` array. Any thoughts? Do I simply perform another API call in getStaticProps(), i.e., have 2 separate calls to the same endpoint?

```
export async function getStaticPaths() {
  // Call an external API endpoint to get all journals
  const res = await fetch('http://localhost:3000/api/journals')
  let allJournals = await res.json()

  // extract their name, slugs AND their path (i.e., including the /j/)
  let paths = allJournals.map((journal) => ({
    params:{name: journal.name, id: journal.slug }
  }))

  return { paths, fallback: false } // Only "id" is returned here.
}
```

Best wishes, and stay safe,

Jack

**Comment by balazsorban44** [comment]
You may not have control over the api you are calling, but if you do, you could simply create another endpoint that only returns paths? I don't think it is possible to return other than the path from getStaticPATHS. If you need more than the name in getStaticProps, than it wouldn't add too much overhead to get the name twice.

But I see your point. It may feel a bit unnecessary to fetch things twice. You can actually try fallback: true, that would create less load on the api on build time, while also reducing the build time itself.

**Comment by revskill10** [comment]
I also need this. One reason is that, currently you can't have `getStaticPaths` without `getStaticProps`, so why not allowing additional props returning by `getStaticPaths` to be consumed in `getStaticProps`. The benefit is clear (only one API call for all we need in both methods), as well as keeping backward compatible.

**Comment by JCKalt** [comment]
I'm curious... what does return { paths, fallback: false} return ? this is not a valid JSON object because paths does not have a key.
